clients/src/main/java/org/apache/kafka/common/metrics/MetricConfig.java

```diff
@@ -43,6 +43,15 @@ public MetricConfig() {
         this.recordingLevel = Sensor.RecordingLevel.INFO;
     }
 
+    public MetricConfig(MetricConfig config) {
+        this.quota = config.quota;
+        this.samples = config.samples;
+        this.eventWindow = config.eventWindow;
+        this.timeWindowMs = config.timeWindowMs;
+        this.tags = config.tags;
```

Comment:
Being picky, this copy constructor is not copying the fields from the input parameter. Shouldn't we create a copy of the quota and tags, for example?